File: solutions/dart/dnd-character/1/lib/dnd_character.dart
import 'dart:math';

class DndCharacter {
  int strength = 0,
      dexterity = 0,
      constitution = 0,
      intelligence = 0,
      wisdom = 0,
      charisma = 0;
  int hitpoints = 0;

  static DndCharacter create() {
    var c = DndCharacter();
    c.strength = DndCharacter.ability();
    c.dexterity = DndCharacter.ability();
    c.constitution = DndCharacter.ability();
    c.intelligence = DndCharacter.ability();
    c.wisdom = DndCharacter.ability();
    c.charisma = DndCharacter.ability();
    c.hitpoints = 10 + DndCharacter.modifier(c.constitution);

    return c;
  }

  static int ability() {
    final rnd = new Random();
    return rnd.nextInt(15) + 3;
  }

  static int modifier(int constitution) {
    return ((constitution - 10) / 2).floor();
  }
  // Put your code here
}

File: solutions/python/dnd-character/1/dnd_character.py
import random

ABILITIES=('strength', 'dexterity', 'constitution', 'intelligence', 'wisdom', 'charisma')

class Character:
    def __init__(self):
        for ability in ABILITIES:
            setattr(self, ability, self.ability())
        
        self.hitpoints=10+modifier(self.constitution)

    def ability(self):
        dices=sorted(random.randint(1,6) for _ in range(4))
        return sum(dices[1:])

def modifier(value):
    return (value - 10) // 2
